railgun/internal/ctrlutils/ingress-status.go

```diff
@@ -59,7 +59,11 @@ func PullConfigUpdate(ctx context.Context, kongConfig sendconfig.Kong, log logr.
 		case updateDone := <-kongConfig.ConfigDone:
 			log.V(4).Info("receive configuration information. Update ingress status %v \n", updateDone)
 			wg.Add(1)
-			go UpdateIngress(ctx, &updateDone, log, cli, kiccli, &wg, ips, hostname, kubeConfig)
+			go func() {
+				if err := UpdateIngress(ctx, &updateDone, log, cli, kiccli, &wg, ips, hostname, kubeConfig); err != nil {
+					log.Error(err, "failed to update resource statuses")
```

Comment:
If we dont swallow any error, please panic on error messages.

Reply:
Are you sure? That would crash the entire controller for a single status update. You wrote this code originally and the current behavior _is to swallow errors_. Do you think we should continue that (but log)? I don't think panic is correct here because there legitimate and transient reasons why status updates might fail.

Reply:
well, theoretically, if user log golang error, needs action. The original mechanism is using retry cover error - if fails in the end, tc/user would retry configuring ingress.

Since you purposely catch error, the rule is to act - either panic or retry again, does it make sense ?

Reply:
Yes what you're saying makes sense. This is why I was originally hoping we could put the status updates into a controller, and why I was interested in retry logic for failed status updates in the original PR.

How about this: would you be OK with me creating a follow up issue to make the changes necessary to handle this error in a follow up PR, but for now we just add the log entry so that:

1. lint is satisfied
2. we're no longer silently dropping errors

If so, I can create an issue and make it a blocker for 2.0 and get that solved asap.

Reply:
Sure, the log entry is fine.

Reply:
Sounds good, here's the follow up: https://github.com/Kong/kubernetes-ingress-controller/issues/1546